reactComponent/src/index.fs

```diff
@@ -0,0 +1,28 @@
+module ReactComponent
+
+open Fable.React
+
+// Since the props object is created from Javascript it's best to be defensive (use options) for the top level attributes.
+// If you used a string it might work but there is no check to make sure the caller provided a value.
+type Props = 
+    abstract member name : string option
+
+// A standard react component dervies from either Fable.React.Component or PureComponent.
+// This is the most verbose way to define a component but also the most flexible.
+type StandardComponent(props) as self=
+    inherit PureComponent<Props,unit>(props) with
+    do self.setInitState()
+    override __.render() =
+        let name = defaultArg props.name "nobody"
+        div [] [str (sprintf "Hello %s" name)]
+
+// If you expect the component to be consumed from javascript with JSX you can expose a bare function as a react component and that will work (including hooks).
+// If you call this from F# it would act as a plain function so you would have to wrap it with FunctionComponent.Of if you need hooks to work. 
+// see: https://fable.io/blog/Announcing-Fable-React-5.html
+let FunComponent (props:Props) =
+    let name = defaultArg props.name "nobody"
+    div [] [str (sprintf "Hello %s" name)]
+
+// Using FunctionComponent.Of also works and gives you a consistent behavior if called from either Javascript or Fable
+let Fable5FunComponent (props:Props) =
+    FunctionComponent.Of(FunComponent, memoizeWith = equalsButFunctions) props

```

Comment:
In order for memoization to work this needs to be a value, not a function (just remove the `props` parameter):

```fsharp
let Fable5FunComponent =
    FunctionComponent.Of(FunComponent, memoizeWith = equalsButFunctions)
```

Also, unfortunately the return value of `FunctionComponent.Of` cannot be used in JSX. You need a slightly modified variant for that: [example](https://fable.io/repl/#?code=LYewJgrgNgpgBAJRgQwMYBcDCJgAcQB2MB6AUKSLsXAGLIBGsAdNgE4wVUG0PNsxMAUgGcAkiRitKnanUYCkaMqQD0KuABEYqKMnZwAZhAIYAloThgYB00TBwA5lBD1kUKAE84tuOgAW8AASACoAsgAycAA8wqisprjoAHy+yA6ksOhwANpRAOLOrlBJALpwoMboAIK4uHAAFKg4AFyIKBgAorDAxOjBHlRRAOQAClK4wkkAlK3GplkAvHAAVsIAcsjopgBuHKrqNMZmhNh4hL1MAPIG3sLeEgRW9uggcPTwEMIw9gZSwLQAYn2cGCrxgAA98F9vFkfIIAMpwADu8CI3zgyDgwigpgcfnQnnK4FMNnR2z0pmQJGB9QA6vBhH4QNB7DiANYwQnIMD2ea+V7+eByZiKDBwHH0Vh6DxTUjoAZCo5bE44fBEEjI+Z+UhwXVY9CbUyocowYDvVi0ECsBE6vV2+16+rsR6SVqjcbCAC0SVF6C6pt6ABpbQ7Q7qAPxgUzCXC6DwbHqtYToeIEBzBsOZ8M9UCmABeMFpWrdY0od29cHdZbgFfoIBAUFlmdDrV9-p6JH6gyrExSCxD9uAm1QfhNuYLRf8mv8A7tAB84PCcPA9DAOgBHCBuGtJWehzJjkDt3pwJa+gBCtijaeETF9TBzICdxCsrEDGPYG63jb3DqH6BHSxo1jZB42QHpp21ZtdQXJcIIIcD4ArR9jxISNgLjBN4CiT04AAIlCU0nzwuAAGo4AQiDyLwqY8N-e0FzWc4dwaJtoJQ7penoxjmIrYw63BOBnVfcg1DgDpHjgEAbjrUxYFYED0BgABCOBwhgdAhjuZM9CyKwbAIWwHCkiALSaM51WUMT4VsVB4EFOBcA9KT6GWbRYTuOIUCUn4-jgQRkHJWJ4kSGEtLeGBk35CLLGsYhhB2eB6k+eBKGVAhhCmQwrV8AJ+TqWBdigDF0BTUx6AgJTb2BUQbg8Zk4BS+xMWTVNjL5YBcXxZErTZN5KtyyR4GjCjXhHbQ+pecpkA5LFTPsvLUDceTHKkbZTCeDE4HJKAIAEOUFTgUsJlPOABwYVqlDHc0KMQuAkzKtMpMScwCFE9RargeqIDgCEqDFBzzLVE8pveOAmgyiAel8nAVkC5BgoSLIkS1fz4QADS+hqlu4P6QGhTFXH0IwTHSjE7kxdgrqB84NSpZ4-E2TV3B61g+vqWzduvYymRANlMqYGq6ux5bcpG35YZoAEYR6lkMTFBGtpAnwSeObhhFeb7ZagexGd2aKkSlOo+RRqdDlJ17TmBkgrhueocoAA2k821YdrKSSxn60V1+t+YN3qmDOsSvhgVp8XQCZmjUAxeAEcwVEYEAHBUKoCAIZkTCMz1hRgT1fU9ABWJh8WAKAMg02hjCt2msnqJyy2aY7MtPAcD0o+Aln05BoGqVhjPriYmHb-D07rMAPDou0o22HIymyVqGhjVN0BuPDAk5Zw4AAUmEEj26mEp3rgABVBKnpd9Lq8spgaCtBFkXgHGMR5Z+o3S7d3j18xTPFUxZsffMhoLA5UxKgT4Lx-hRhjHGW6PRy5ZBzgXc2V8Tz9jtBfS2qoa43zvvCKITddxhnqMgrBlkMxhgAROVGSwYCbjcMIc8lUMGEGEOQ0MUCQJgQgksPCOcUEkBokfeEL5Brg1ISDV498DA5QqCQIyMJSCyOqLUHg8gkFV3EdSIAA&html=DwCwLgtgNgfAUHUBTAhgE3gAm54ElgqbhgAOAtEgI4CuAlgG4C8A5AMID2AdmEj+QBUAnqSQtMAY268erXgA8wAenDQA3JJAoATgGcCTGmABm5ABwssOYLona6pMJl3aJTAEQB9TwAkA8gDKAt5KUHQARrpK2qgSYAB0pNocaDRxdNzxEHRc8QBWuu4wwEq29o5W2DZ2Dk4ubl6+gcGeoRFRMShx5GgcEInJqemZ2bkFRSVltZW4UxVwSkqYAKpcxhzaYDRcKLxQQpgA7kiSKFxcHE40+pgAUgEAGs5CPCjyxEgxC0sA6ieHHBoUDQmC4SCQIKISSQSQ4EiQul0G0w620mDAWjAcGM22GXEwEEBPAAgqRSAAKABiKHCUCQnAgpG4fDAAEpMABvOA4TAAJViYAAIn4ALLxGJcNCfcncnk4Ra4am0+l9JlgniglD4DxKukAQncmCUMx5-K6CTsqF4AFE6fgeFSaXSGWqWQAaTma-AALkw7l1SANmAAvqy3bK5ZhehIaPaEgBzAi2pBxgBCQgAkmhye4UGT3KzWXBgxHcKUavMSiBUBgEMBwikDhIoChER48xQpK8cp8iqXgGhGJg6Gh2-nJC227n88UlIOGPASg20EJ4IgVJBYHAgA&css=BYFwtgNgNAUARgewCYE8AEBvGadoGYIB2IAtAM4CWAXgKYBcaAjAGwAOAHgNwwC+MMAOjABDCoRIBjIiFGEaAJ0zZcSCmVYRhKBngg0uynAHcKSEMAaMADFYCk3XGmA0KAc1CWb9w2gBWAVzIQCjwUSWkaYgYJSJAFB1xhCDdxCjiwMmjY+J9dfRJVeRoJYKJohAh-MEJuPkExVn8QJUdWYSRVQlcGAQAmAFYisAScAmJyanomNgNHEzMLNAGrDhG0EXlXMRJEEBAEMEsh2pggA).

Reply:
Ahh! I think the two errors work together.  In this case <Fable5FunComponent name='x' /> does work from JSX but it's probably working *because* it's a function instead of a value.  If I look at it in the react tools I see:
![image](https://user-images.githubusercontent.com/2947959/58095202-7afe9800-7ba0-11e9-9abd-c40947126e98.png)

So there's an extra component in there.  I don't know enough about react to know if this would actually work correctly with memo. I guess either way it would not work for a Fable caller.  Updated PR incoming!